static/profile/js/profile.js

```diff
@@ -0,0 +1,486 @@
+  
+  var mongoprofiles = [];
+
+  // Fetch data from mongo
+  $('#pe_status').hide().text('Loading profile records ...').fadeIn('slow');
+  $.ajax('/api/v1/profile.json', {
+    success: function (records) {
+      c_profile = {};
+      mongoprofiles = records;
+      // create new profile to be edited from last record
+      if (records[0] && records[0].dia) {
+        // Use only values(keys) defined in defaultprofile, drop the rest. Preparation for future changes.
+        c_profile = _.cloneDeep(defaultprofile);
+        for (var key in records[0]) {
+          if (records[0].hasOwnProperty(key)) {
+            if (typeof c_profile[key] !== 'undefined') {
+              c_profile[key] = records[0][key];
+            }
+            // copy _id of record too
+            if (key === '_id') {
+              c_profile[key] = records[0][key];
+            }
+          }
+        }
+        convertToRanges(c_profile);
+        
+        $('#pe_status').hide().text('Values loaded.').fadeIn('slow');
+        mongoprofiles.unshift(c_profile);
+      } else {
+        c_profile = _.cloneDeep(defaultprofile);
+        mongoprofiles.unshift(c_profile);
+        $('#pe_status').hide().text('Default values used.').fadeIn('slow');
+      }
+    },
+    error: function () {
+      c_profile = _.cloneDeep(defaultprofile);
+      mongoprofiles.unshift(c_profile);
+      $('#pe_status').hide().text('Error. Default values used.').fadeIn('slow');
+    }
+  }).done(initeditor);
+  
+  // convert simple values to ranges if needed
+  function convertToRanges(profile) {
+    if (typeof profile.carbratio !== 'object') { profile.carbratio = [{ 'time': '00:00', 'value': profile.carbratio }]; }
+    if (typeof profile.sens !== 'object') { profile.sens = [{ 'time': '00:00', 'value': profile.sens }]; }
+    if (typeof profile.target_low !== 'object') { profile.target_low = [{ 'time': '00:00', 'value': profile.target_low }]; }
+    if (typeof profile.target_high !== 'object') { profile.target_high = [{ 'time': '00:00', 'value': profile.target_high }]; }
+    if (profile.target_high.length !== profile.target_low.length) {
+      alert('Time ranges of target_low and target_high don\'t  match. Values are restored to defaults.');
+      profile.target_low = _.cloneDeep(defaultprofile.target_low);
+      profile.target_high = _.cloneDeep(defaultprofile.target_high);
+    }
+  }
+  
+  function initeditor() {
+    // Load timezones
+    $('#pe_timezone').empty();
+    moment.tz.names().forEach(function addTz(tz) {
```

Comment:
you can do this small changes on merge. i'm unable to check it with codacy as it takes always 5h+